File: vigenere.py
def prepare_string(string: str) -> str:
    # deleting non-alphabetic characters and turning to uppercase
    return ''.join(c.upper() for c in string if c.isalpha())

def encrypt(plaintext: str, keyword: str):
    """encrypting text using vigenere encryption"""
    # preparing texts
    plaintext = prepare_string(plaintext)
    keyword = prepare_string(keyword)
    keyword = [ord(x) - 64 for x in keyword]

    encrypted =  ''
    key = 0

    for char in plaintext:
        # If keyword ends go to the begin
        if key == len(keyword): 
            key = 0
        
        # Appending encrypted letter
        encrypted += encrypt_letter(char, keyword[key])

        # Select next key for shifting
        key += 1

    # Return encrypted text or None if there's none
    return encrypted if encrypted != '' else None

def encrypt_letter(char, shift):
    # Return shifted character for encryption
    return chr((ord(char) + shift - 65) % 26 + 65) 

def decrypt(text: str, keyword: str):
    """decrypting text using vigenere decryption knowing they key"""
    # preparing texts
    text = prepare_string(text)
    keyword = prepare_string(keyword)
    keyword = [ord(x) - 64 for x in keyword]

    decrypted =  ''
    key = 0

    for char in text:
        # If keyword ends go to the begin
        if key == len(keyword): 
            key = 0
        
        # Appending decrypted letter
        decrypted += decrypt_letter(char, keyword[key])

        # Select next key for shifting
        key += 1

    # Return decrypted text or None if there's none
    return decrypted if decrypted != '' else None

def decrypt_letter(char, shift):
    # Return back shifted character for decryption
    return chr((ord(char) - shift - 65) % 26 + 65) 

File: vigenere.cpp
#include <iostream>
#include <string>

std::string encrypt(std::string plaintext, std::string keyWord)
{
	std::string encrypted = "";
	int shift { 0 };

	for (unsigned int i { 0 }, k { 0 }; i < plaintext.length(); i++, k++)
	{
		// Skipping spaces
		while (plaintext[i] == char(32)) i++;

		// Going to the start of the key when it ends
		if (k >= keyWord.length())
			k = 0;

		// Converting key into shift numbers
		if (isupper(keyWord[k])) 
			shift = int(keyWord[k]) - 64;
		else 
			shift = int(keyWord[k]) - 96;

		// Encrypting plaintext
		if (isupper(plaintext[i]))
			encrypted += char(int(plaintext[i] + shift - 65) % 26 + 65);
		else
			encrypted += char(int(plaintext[i] + shift - 97) % 26 + 97);
	}
	
	return encrypted;
}

int main()
{
	// Enter text
	std::string text = "";
	std::cout << "Enter the text to be encrypted: ";
	std::getline(std::cin, text);

	// Enter key
	std::string key = "";
	std::cout << "Enter the encryption key: ";
	std::cin >> key;

	// Display encrypted text
	std::cout << "Encrypted text: " << encrypt(text, key);

	return 0;
}